feature-pitches/010-compute-sessions/compute-sessions.md

```diff
@@ -0,0 +1,220 @@
+
+#### Session
+
+A session is an instantiation of an *environment* in a specific project. A user can start a session and have live, interactive access to the project and its code and storage.
+
+Sessions can have the following statuses:
+
+- ⬜ Inactive
+- 🕗 Launching
+- 🟢 Running
+- ⏸ Paused
+- ⚠️ Failed/Error
+
+When an *environment* is added to a project, it automatically becomes a project *session* in the state *inactive*.
+
+#### Session Configuration
+
+The session configuration customizes a session for an environment in a project. Via the session configuration, a user can:
+
+- set minimum compute resource requirements
+- [nice to have] select which of the project’s code repos to mount in the session [default: mount all]
+- [for the future/out of scope] select which of the project’s storages to mount in the session [default: mount all]
+- customize the name of the session
+
+When a user wants to customize the resources of their session (compute resources, repos or storages mounted), they modify the *session configuration*.
+
+Session configuration is project-specific.
+
+#### Session Front-End [for the future/out of scope]
+
+In Renku 1.0, we would like to offer that a user can swap out different front ends to their sessions, independently of the session environment. So, a user could find a data science environment on Renku that they want to reuse, and but use VSCode instead of Jupyter as the front end for the session without changing the rest of the environment.
+
+This is something we want in Renku 1.0, but is out of scope for this build for the sake of time.
+
+### User Flows
+
+#### User Flow: Add a Session to a Project
+
+![breadboard-add-session-pt-1.jpg](breadboard-add-session-pt-1.jpg)
+
+continuing to the right...
+
+![breadboard-add-session-pt-2.jpg](breadboard-add-session-pt-2.jpg)
+
+A user has 4 options for adding a session to their project. Only the first 2 are considered critical for this build. These options are listed in order of priority, so if this build doesn’t get to implementing all of them, that’s ok.
+
+##### Priority 1: Select from the RenkuLab Default Environments
+
+- These are built and maintained by us
+- **At a minimum, 1 default environment must be created as part of the build!** (a Jupyter Python environment)
+- These images are pre-built, so launching  a session from one is fast (the user doesn’t have to wait for an image build)
+- [nice to have] The default environments should be configured via the admin panel
+
+##### Priority 2: Use an external image
+
+- There are a whole bunch of restrictions for what must be in the image for it to work on RenkuLab. We will have to document these limitations for users.
+  - We *think* that any image coming from the Jupyter stack should work
+  - ... Or at least Renku base images
+- Only publicly accessible images are supported
```

Comment:
This should either say "... will be supported as a part of this build, and authentication options added later." _or_ add a "nice to have" to deal with authentication.